File: easy/3633-earliest-finish-time-for-land-and-water-rides-i/solution.java
/**
 * 3633. Earliest Finish Time for Land and Water Rides I
 * Time: O(n * m)
 * Space: O(1)
 */
class Solution {
    public int earliestFinishTime(
        int[] landStartTime,
        int[] landDuration,
        int[] waterStartTime,
        int[] waterDuration
    ) {
        int answer = Integer.MAX_VALUE;

        for (int land = 0; land < landStartTime.length; land++) {
            for (int water = 0; water < waterStartTime.length; water++) {
                int landFirst = Math.max(
                    landStartTime[land] + landDuration[land],
                    waterStartTime[water]
                ) + waterDuration[water];

                int waterFirst = Math.max(
                    waterStartTime[water] + waterDuration[water],
                    landStartTime[land]
                ) + landDuration[land];

                answer = Math.min(answer, Math.min(landFirst, waterFirst));
            }
        }

        return answer;
    }
}

File: easy/3633-earliest-finish-time-for-land-and-water-rides-i/solution.py
from typing import List


class Solution:
    def earliestFinishTime(
        self,
        landStartTime: List[int],
        landDuration: List[int],
        waterStartTime: List[int],
        waterDuration: List[int],
    ) -> int:
        """
        3633. Earliest Finish Time for Land and Water Rides I
        Time: O(n * m)
        Space: O(1)
        """
        answer = float("inf")

        for land_start, land_duration in zip(landStartTime, landDuration):
            for water_start, water_duration in zip(waterStartTime, waterDuration):
                land_first = max(land_start + land_duration, water_start) + water_duration
                water_first = max(water_start + water_duration, land_start) + land_duration
                answer = min(answer, land_first, water_first)

        return answer
